.github/workflows/build.yml

```diff
@@ -121,9 +121,42 @@ jobs:
             fetch-upstream: "true"
       fail-fast: false
     steps:
+      - run: |
+          echo "OS_ARCH=`uname -m`" >> $GITHUB_ENV
       - uses: actions/checkout@v3
+      # Cache `cargo-make`, `cargo-cache`, `cargo-sweep`
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo
+          key: ${{ hashFiles('.github/workflows/cache.yml') }}-${{ runner.os }}-${{ env.OS_ARCH }}
+      # Cache first-party rust code crate dependencies
+      - uses: actions/cache@v3
+        with:
+          path: |
+            .cargo
+          key: ${{ hashFiles('.github/workflows/cache.yml') }}-${{ runner.os }}-${{ env.OS_ARCH }}-${{ hashFiles('sources/Cargo.lock') }}-${{ hashFiles('.github/workflows/build.yml') }}-${{ github.head_ref }}
```

Comment:
```
${{ github.head_ref }}
```

as part of the hash key seems redundent. If the action runs, and creates a cache with key:
```
{{ runner.os }}-${{ env.OS_ARCH }}-${{ hashFiles('sources/Cargo.lock') }}
```
 it'll create a cache general use cache that is associated with:
- that runner's os
- that runner's arch
- the hash from the first party code dependencies

If we need to run that action again, it'll hit that same cache (since rerunning it, non of those 3 variables should have changed).

I believe simplifiying the keys will also cause fewer caches to be created in general since we won't have to worry about arbitrary files or variable (like in those files in `.github`). Instead, general use caches will be generated that can be used across many different runs or many different PRs.

Reply:
> I believe simplifiying the keys will also cause fewer caches to be created

There is no limit in the number of caches we can have at a time. By default caches that aren't used over 10 days or something gets purged. We do have a limit on total cached size, around 10 GB.

Reply:
Right, that scenario is the desired behavior. 

> So wouldn't the hash for that lockfile be unique to that PR's branch and cause a new cache to be created?

I guess that's true, if the PR changes the lockfiles, it's guaranteed to cache-miss on the initial build workflow and have to generate their own. That's also the reason why I split it into 3 cache actions so some parts can still use the cache even if only `sources/Cargo.lock` is changed.

We'd still end up with the same amount of caches regardless if we append the `github_ref` or not. Maybe there's a case to be made for weird PR chains like `"pr1" -> "pr2" -> "develop"`. I'll remove the `github_ref` from the key.